### PR TITLE
Adjust the Travis CI matrix: PHP 7.0 and 7.1, fewer revisions of libmemcached

### PR DESCRIPTION
**Pull request by sodabrew**

### CODE REVIEW

**Comment by sodabrew** [comment]
I actually don't know if the PHP 7 branch is capable of working with PHP 5.5 or 5.6. My hope would be to have a single php-memcached master branch and next release that works across the three supported revisions. Let's see how it goes.

**Comment by sodabrew** [comment]
With PHP < 7.0
```
In file included from /home/travis/build/php-memcached-dev/php-memcached/msgpack-php/msgpack.c:16:0:
/home/travis/build/php-memcached-dev/php-memcached/msgpack-php/php_msgpack.h:4:56: fatal error: Zend/zend_smart_str.h: No such file or directory
```

**Comment by TysonAndre** [comment]
The php7 branch targets php7 only. It won't compile under any version of php5.

If they are planning to have a master branch that targets master and php7, it is possible two ways (But they probably wouldn't do that until php7 was stable):

1. With a lot of macros and ifdefs (Doable for simple extensions that don't build complicated data structures. Makes understanding the code or writing php7 specific optimizations, such as switching to zend_string slightly harder in the future)
2. With separate source folders - This is what igbinary and solr do (E.g. https://github.com/igbinary/igbinary/pull/62 )

- Move the php5 source code to a subfolder src/php5/
- Copy the source from the php7 branch to src/php7/
- Update package.xml
- Update src/php5 with the new features/api changes in php7.
- Update tests to work for both branches

**Comment by sodabrew** [comment]
I think the sane path forward is:
* Branch master to a `stable-2.x` that works with PHP 5.x. At this time, only PHP 5.6 remains supported upstream.
* Merge php7 to master (possibly rename master as `stable-2.x` and rename php7 as master).
* Add support for PHP 7.1 to the future php7 / master branch, since PHP 7.1 is in RC already.

File: .travis.yml [hunk 1]
```diff
@@ -1,21 +1,14 @@
...
-    - LIBMEMCACHED_VERSION=1.0.7
-    - LIBMEMCACHED_VERSION=1.0.6
-    - LIBMEMCACHED_VERSION=1.0.2
-    #- LIBMEMCACHED_VERSION=0.53
-    #- LIBMEMCACHED_VERSION=0.49
-    #- LIBMEMCACHED_VERSION=0.44
+    - LIBMEMCACHED_VERSION=1.0.18 # Debian Jessie / Ubuntu Xenial
+    - LIBMEMCACHED_VERSION=1.0.16 # RHEL / CentOS 7
+    - LIBMEMCACHED_VERSION=1.0.8  # Debian Wheezy / Ubuntu Trusty
+    - LIBMEMCACHED_VERSION=0.44   # Ubuntu Precise
```
Comment (andrerom) [inline .travis.yml]: Is Precise still relevant for 3.0 (php7) code base? Unless it's meant to supported, and given tests fails with it, I wouldn't re-add it here.

**Comment by sodabrew** [comment]
PHP 7.1 compat for msgpack blocked on https://github.com/msgpack/msgpack-php/issues/100

**Comment by sodabrew** [comment]
@laruence Ideally a fix for msgpack/msgpack-php#100 will land first so that I can remove my personal branch from this PR, if you have time could you merge msgpack/msgpack-php#104?

**Pull request merged**